### PR TITLE
Nested, re-used fragment query issue

### PR DESCRIPTION
**Pull request by benmccallum**
Addresses #3166

### CODE REVIEW

File: src/HotChocolate/Core/HotChocolate.Core.sln [hunk 1]
```diff
@@ -61,33 +61,35 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "HotChocolate.Execution.Benc
...
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "HotChocolate.StarWars", "benchmark10\StarWars\HotChocolate.StarWars.csproj", "{0BE70B30-33CC-4BB7-8EEA-4981A2E9417C}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HotChocolate.Types.Scalars", "src\Types.Scalars\HotChocolate.Types.Scalars.csproj", "{8B7EEF6D-DFF2-4D7B-9E5D-6CA34A8BC260}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "HotChocolate.Types.Scalars", "src\Types.Scalars\HotChocolate.Types.Scalars.csproj", "{8B7EEF6D-DFF2-4D7B-9E5D-6CA34A8BC260}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HotChocolate.Types.Scalars.Tests", "test\Types.Scalars.Tests\HotChocolate.Types.Scalars.Tests.csproj", "{480E09E8-A0C4-49A7-AAC4-4AA5EC0733C5}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "HotChocolate.Types.Scalars.Tests", "test\Types.Scalars.Tests\HotChocolate.Types.Scalars.Tests.csproj", "{480E09E8-A0C4-49A7-AAC4-4AA5EC0733C5}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "HotChocolate.AspNetCore", "..\AspNetCore\src\AspNetCore\HotChocolate.AspNetCore.csproj", "{70AB2443-3FE1-4C65-BBB3-4A57872CC2A4}"
```
Comment (benmccallum) [inline src/HotChocolate/Core/HotChocolate.Core.sln:92]: Needed to add this project to the sln to be able to run the tests...

VS did it's thing though and changed a bunch of other stuff...

File: src/HotChocolate/Core/test/Execution.Tests/Integration/StarWarsCodeFirst/__snapshots__/StarWarsCodeFirstTests.NestedFragmentsWithNestedObjectFieldsAndSkip.snap [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+              }
+            }
+          },
+          {
+            "node": {
+              "__typename": "Human",
+              "name": "Leia Organa",
+              "otherHuman": {
+                "name": "Luke Skywalker"
+              }
```
Comment (benmccallum) [inline src/HotChocolate/Core/test/Execution.Tests/Integration/StarWarsCodeFirst/__snapshots__/StarWarsCodeFirstTests.NestedFragmentsWithNestedObjectFieldsAndSkip.snap:21]: If run in BCP against starwars sample aspnet app, it returns `"otherHuman": {}`

**Comment by michaelstaib** [comment]
@benmccallum how did you get it to fail? I have written a test on the server and it still passes.

File: src/HotChocolate/AspNetCore/test/AspNetCore.Tests/HttpPostMiddlewareTests.cs [hunk 3]
```diff
@@ -778,5 +778,73 @@ id @export
...
         }
+
+        [Fact]
+        public async Task Fragments()
+        {
+            // arrange
+            TestServer server = CreateStarWarsServer();
+
+            // act
+             ClientQueryResult result =
```
Comment (michaelstaib) [inline src/HotChocolate/AspNetCore/test/AspNetCore.Tests/HttpPostMiddlewareTests.cs:789]: This now uses a full test server and a POST request.

Comment (benmccallum) [reply src/HotChocolate/AspNetCore/test/AspNetCore.Tests/HttpPostMiddlewareTests.cs:789]: As discussed in Slack, I think this is probably unnecessary now, as my commit 714c8b1 corrects the test. It was `if: true` that causes the problem. With that in place you should be able to repro with just my test. Sorry, fail on my part !

**Comment by michaelstaib** [comment]
this one is replaced by #3328

**Pull request closed without merging**